src/main/java/com/avispl/symphony/dal/infrastructure/management/philips/huebridge/PhilipsHueDeviceCommunicator.java

```diff
@@ -0,0 +1,4123 @@
+						locationList.add(location);
+					}
+				}
+			}
+			if (locationList.isEmpty()) {
+				throw new ResourceNotReachableException("Error creating device automation cannot be empty. Please select the device to create automation");
+			}
+			autoConfiguration.setLocation(locationList.toArray(new Location[0]));
+		}
+
+		//Extract device list by type is room
+		if (TypeOfDeviceEnum.ROOM.getName().equals(type)) {
+			Map<String, String> mapOfDevice = typeAndMapOfDeviceAndValues.get(PhilipsConstant.ROOM);
+			if (mapOfDevice != null) {
+				Map<String, Group> nameOfGroupMap = new HashMap<>();
+				for (Entry<String, String> deviceEntry : mapOfDevice.entrySet()) {
+					Group group = new Group();
+					if (!StringUtils.isNullOrEmpty(deviceEntry.getValue()) && !PhilipsConstant.NONE.equals(deviceEntry.getValue())) {
+						String idDevice = roomNameAndIdMap.get(deviceEntry.getValue());
+						Optional<RoomAndZoneResponse> room = roomList.stream().filter(item -> item.getId().contains(idDevice)).findFirst();
+						if (room.isPresent()) {
+							if (nameOfGroupMap.size() == 0) {
+								group.setId(room.get().getId());
+								group.setType(PhilipsConstant.ROOM.toLowerCase(Locale.ROOT));
+								nameOfGroupMap.put(room.get().getMetaData().getName(), group);
+							} else {
+								Group groupAndDevice = nameOfGroupMap.get(room.get().getMetaData().getName());
+								if (groupAndDevice == null) {
+									group.setId(room.get().getId());
+									group.setType(PhilipsConstant.ROOM.toLowerCase(Locale.ROOT));
+									nameOfGroupMap.put(room.get().getMetaData().getName(), group);
+								}
+							}
+						}
+					}
+				}
+				for (Entry<String, Group> deviceList : nameOfGroupMap.entrySet()) {
+					Location locationItem = new Location();
+					locationItem.setGroup(deviceList.getValue());
+					locationList.add(locationItem);
+				}
+				if (locationList.isEmpty()) {
+					throw new ResourceNotReachableException("Error creating room for automation cannot be empty. Please select the room to create automation");
+				}
+				autoConfiguration.setLocation(locationList.toArray(new Location[0]));
+			}
+		}
+
+		//Extract device list by type is Zone
+		if (TypeOfDeviceEnum.ZONE.getName().equals(type)) {
+			Map<String, String> mapOfDevice = typeAndMapOfDeviceAndValues.get(PhilipsConstant.ZONE);
+			if (mapOfDevice != null) {
+				Map<String, Group> nameOfGroupMap = new HashMap<>();
+
+				for (Entry<String, String> deviceEntry : mapOfDevice.entrySet()) {
+					Group group = new Group();
+					if (!StringUtils.isNullOrEmpty(deviceEntry.getValue()) && !PhilipsConstant.NONE.equals(deviceEntry.getValue())) {
+						String idDevice = zoneNameAndIdMap.get(deviceEntry.getValue());
+						Optional<RoomAndZoneResponse> room = zoneList.stream().filter(item -> item.getId().contains(idDevice)).findFirst();
+						if (room.isPresent()) {
+							if (nameOfGroupMap.size() == 0) {
+								group.setId(room.get().getId());
```

Comment:
same as the comment above - the room.get() could be either extracted into a variable, or the approach could be changed slightly to use Optional type features (instead of using isPresent -> ifPresentOrElse, etc)

Reply:
I fixed it